### PR TITLE
Official cRO support

### PR DESCRIPTION
**Pull request by alisonrag**
add support to Chinese server
* dont need to hardcode password
* poseidon support to cRO_2017-12-13a client
* dont need to execute another poseidon, just the default from openkore
* updated charblocksize 
* updated tables (recvpackets/shuffle/sync/ro txt files)

special thanks to @yon2kong i get some informations from his repository
thanks @sctnightcore  for the account

screenshot:
openkore poseidon
poseidon

to poseidon work just use the latest client and go to openkore\control\poseidon.txt and set this:
```
server_type=cRO_2017-12-13a
CharBlockSize=147
```

you can see a tutorial how to use poseidon here:
http://openkore.com/index.php/Poseidon
here:
https://github.com/yon2kong/poseidon
also here:
https://www.youtube.com/watch?v=4hUOe4dVXjk
discord room : https://discord.gg/aWfnKX5

### CODE REVIEW

**Comment by sctnightcore** [comment]
issues #1513

File: src/Poseidon/RagnarokServer.pm [hunk 1]
```diff
@@ -845,37 +879,64 @@ sub SendCharacterList
...
 	# Character Block
 	my $block;
 	
-	# Filling Character 1 Block
-	$cID = $charID;	$hp = 10000; $maxHp = 10000; $sp = 10000; $maxSp = 10000; $hairstyle = 1; $level = 99; $headTop = 0; $hairColor = 6;
-	$name = "Poseidon"; $str = 1; $agi = 1; $vit = 1; $int = 1; $dex = 1; $luk = 1;	$exp = 1; $zeny = 1; $jobExp = 1; $jobLevel = 50; $slot = 0; $rename = 0;
-	
-	# Preparing Character 1 Block
-	$block = pack($packstring,$cID,$exp,$zeny,$jobExp,$jobLevel,$opt1,$opt2,$option,$stance,$manner,$statpt,$hp,$maxHp,$sp,$maxSp,$walkspeed,$jobId,$hairstyle,$weapon,$level,$skillpt,$headLow,$shield,$headTop,$headMid,$hairColor,$clothesColor,$name,$str,$agi,$vit,$int,$dex,$luk,$slot,$rename);
+	if ($self->{type}->{$config{server_type}}->{charListPacket} eq '0x99d') {
```
Comment (lututui) [inline src/Poseidon/RagnarokServer.pm]: Please review this. It seems to me there's a lot of duplicated code in this if-else

**Comment by agfield** [comment]
Your version also needs to be opened VM？

**Comment by leilei007** [comment]
I still unknown packets for a link to download the complete OP under control

**Comment by sctnightcore** [comment]
@leilei007  unknow packet ?

**Comment by agfield** [comment]
I merged the two versions,now yon2kong'version don't need to grab the password packet.

**Comment by hubertyang88** [comment]
@alisonrag How to run your poseidon in perl cmd line? When I try this in perl cmd line  " E:\okc\src\Poseidon>perl poseidon.pl  ,  I get some error info "Devel::StackTrace version 2 required--this is only version 1.13 at/strawberry/perl/vendor/lib/Exception/Class/Base.pm line 9...."
And  is there any switch can choose port or IP ? such as " /poseidon.pl --char-server=127.0.0.1:6900 --map-server=127.0.0.1:6900"

**Comment by alisonrag** [comment]
@hubertyang88 
to use openkore\src\poseidon\poseidon.pl install this version of perl
http://strawberryperl.com/download/5.12.3.0/strawberry-perl-5.12.3.0.msi

open cmd prompt and type
ppm install Win32::Console

you can start poseidon.pl from a prompt of command:

the options that you have is:
--ragnarokserver_ip=s
--ragnarokserver_port=s
--queryserver_ip=s
--queryserver_port=s
--server_type=s

the default is:
perl poseidon.pl --ragnarokserver_ip=127.0.0.1 --ragnarokserver_port=6900 --queryserver_ip=127.0.0.1 queryserver_port=6900 --servertype=cRO_2017-12-13a

**Pull request merged**